### DISCUSSION TITLE
Help adding --project=compile_commands.json to cppcheck

### DISCUSSION THREAD

**Discussion by tomhampshire** [LazyVim/LazyVim#2117] [Q&A]
I have a simple linting setup in lazyvim, but my lack of lua/lazyvim experience is preventing me from setting my cpp linter up as I need to. 
I need to add `--project=compile_commands.json` to cppcheck's arguments. I have tried to do this as follows:

Below is my `lint.lua` config:
```lua
return {
  {
    "mfussenegger/nvim-lint",
    opts = {
      linters_by_ft = {
        python = { "mypy", "ruff", "pylint" },
        cpp = { "cppcheck" },
        c = { "cppcheck" },
      },
      linters = {
        cppcheck = function()
          local cppcheck = require("lint").linters.cppcheck
          -- I want to add --project=compile_commands.json to cppcheck.args
          cppcheck.args = { "--project=compile_commands.json", unpack(cppcheck.args) }
          return cppcheck
        end,
      },
    },
  },
}

```

I get the following error in nvim:
```

Error executing vim.schedule lua callback: /home/tom/.config/nvim/lua/plugins/lint.lua:14: attempt to index local 'cppcheck' (a function value)
stack traceback:
	/home/tom/.config/nvim/lua/plugins/lint.lua:14: in function 'linter'
	/home/tom/.local/share/nvim/lazy/nvim-lint/lua/lint.lua:193: in function 'lookup_linter'
	/home/tom/.local/share/nvim/lazy/nvim-lint/lua/lint.lua:202: in function 'try_lint'
	.../share/nvim/lazy/LazyVim/lua/lazyvim/plugins/linting.lua:83: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
```

I believe this is due to me using nvim-lint before it is initialised, but I'm not sure how an alternative solution might look. Any guidance would be appreciated.

**Comment by dpetka2001** [comment]
It appears that the function in `linters` table overrides the default one, so you would have to pass the default implementation of `cppcheck` as well. See #1887 for more info and a workaround to come around this problem (hopefully the maintainer will take into consideration that issue and maybe add some kind of `prepend_args` for `nvim-lint` as well so it'll be easier to configure) . I don't use it myself, so can't provide more help.
